wordlespace.html

```diff
@@ -332,50 +37,45 @@
+              if (chr !== greens[i]) {
                 return false;
               }
-            }
-            return true;
-          })
-          // letter by letter check
-          .filter(w => {
-            for (let i = 0; i < 5; i += 1) {
-              let chr = w.charAt(i);
-              // if we have a green letter, it must match
-              if (greens[i] !== '') {
-                // if it doesn't match our green clue, then it fails.  If it does match, go on to the next letter.
-                if (chr !== greens[i]) {
-                  return false;
-                }
-              } else { // no green letter for this position, so look at yellow and gray clues.
-                // this letter can't be any of the yellow letters for this space
-                if (yellows[i].includes(chr)) {
-                  return false;
-                }
-                // this letter can't be a gray letter (unless it's also a yellow)
-                // we check for yellows here because we want the conflict to resolve in favor of
-                // keeping the word in. You can get the letter both ways which indicates that there
-                // is (for example) one of the letter but not two. We don't use this sort of hint
-                // at the moment, but if someone is inputing letters according to our directions,
-                // this will keep it working.
-                if (!allYellows.includes(chr) && grays.includes(chr)) {
-                  return false;
-                }
+            } else { // no green letter for this position, so look at yellow and gray clues.
+              // this letter can't be any of the yellow letters for this space
+              if (yellows[i].includes(chr)) {
+                return false;
+              }
+              // this letter can't be a gray letter (unless it's also a yellow)
+              // we check for yellows here because we want the conflict to resolve in favor of
+              // keeping the word in. You can get the letter both ways which indicates that there
+              // is (for example) one of the letter but not two. We don't use this sort of hint
+              // at the moment, but if someone is inputing letters according to our directions,
+              // this will keep it working.
+              if (!allYellows.includes(chr) && grays.includes(chr)) {
+                return false;
               }
             }
-            return true;
-          });
+          }
+          return true;
+        });
 
       // count letter frequencies and sort by count, highest to lowest
       words.forEach(w => new Set([...w]).forEach(ltr => countLetter(letterFreq, ltr)));
       letterFreq = new Map([...letterFreq].sort(([k1, v1], [k2, v2]) => v2 - v1));
 
       $('#frequency').html(Array.from(letterFreq.entries(), ([letter, count]) => '<li>' + letter + ': ' + count + '</li>').join(''));
+      $('#words').toggle(this.checked);
```

Comment:
hmm, what's the advantage of doing this here?  I assume this goes hand in hand with removing `checked` from the input in the initial HTML.  Is the point that it's hidden until after it's generated the first time?

Reply:
yes. we forgo rendering the whole list until the first clue.